### PR TITLE
[8.0][OAuth 2] dirac-login (AS) added VOMS extension

### PR DESCRIPTION
**Pull request by TaykYoku**
If there are settings in the user group configuration to add a VOMS extension, add it automatically.

BEGINRELEASENOTES

*Framework
NEW: dirac-login (AS) added VOMS extension

ENDRELEASENOTES

### CODE REVIEW

File: src/DIRAC/FrameworkSystem/scripts/dirac_login.py [hunk 1]
```diff
@@ -288,16 +294,29 @@ def loginWithCertificate(self):
 
         # Create local proxy with group
         self.outputFile = self.outputFile or Locations.getDefaultProxyLocation()
-        result = chain.generateProxyToFile(self.outputFile, int(self.lifetime or 12) * 3600, self.group)
+        parameters = (self.outputFile, int(self.lifetime or 12) * 3600, self.group)
+
+        # Add a VOMS extension if the group requires it
+        if (result := chain.generateProxyToFile(*parameters))["OK"] and (result := self.__enableCS())["OK"]:
```
Comment (fstagni) [inline src/DIRAC/FrameworkSystem/scripts/dirac_login.py:300]: Isn't this equivalent to 
```suggestion
        if chain.generateProxyToFile(*parameters)["OK"] and self.__enableCS()["OK"]:
```
?

File: src/DIRAC/FrameworkSystem/scripts/dirac_login.py [hunk 2]
```diff
@@ -288,16 +294,29 @@ def loginWithCertificate(self):
...
+            if not self.group and (result := findDefaultGroupForDN(credentials["DN"]))["OK"]:
+                self.group = result["Value"]  # Use default group if user don't set it
+            # based on the configuration we decide whether to add VOMS extensions
+            if getGroupOption(self.group, "AutoAddVOMS", False):
+                if not (vomsAttr := getVOMSAttributeForGroup(self.group)):
+                    print(HTML(f"<yellow>No VOMS attribute foud for {self.group}</yellow>"))
```
Comment (fstagni) [inline src/DIRAC/FrameworkSystem/scripts/dirac_login.py:306]:
```suggestion
                    print(HTML(f"<yellow>No VOMS attribute found for {self.group}</yellow>"))
```

File: src/DIRAC/FrameworkSystem/scripts/dirac_login.py (hunk 1 above)
Comment (TaykYoku) [reply src/DIRAC/FrameworkSystem/scripts/dirac_login.py:300]: in your case we not have `result` variable

we should handle `result` if it's not `OK`:
```python
       if not result["OK"]:
            return result
```

**Comment by fstagni** [comment]
You have a conflict

**Review by fstagni — APPROVED** [review]

**Pull request merged**